source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -20,10 +20,15 @@ namespace MaterialX
 // ShaderGraph methods
 //
 
-ShaderGraph::ShaderGraph(const ShaderGraph* parent, const string& name, ConstDocumentPtr document) :
+ShaderGraph::ShaderGraph(const ShaderGraph* parent, const string& name, ConstDocumentPtr document, const StringSet& reservedWords) :
     ShaderNode(parent, name),
     _document(document)
 {
+    // Add all reserved words as taken identifiers
+    for (const string& n : reservedWords)
+    {
+        _identifiers[n] = 1;
```

Comment:
Each graph now keeps track of its identifiers in local scope. But here also add in all global reserved words since they are not allowed as identifiers.